solr/solrj/src/java/org/apache/solr/common/SolrDocument.java

```diff
@@ -195,6 +197,19 @@ public Collection<Object> getFieldValues(String name) {
     return null;
   }
 
+  /** Get the value or collection of values for a given field. */
+  public Map<String, Object> getSubsetOfFields(Set<String> fieldNames) {
```

Comment:
I'd suggest we be conservative about adding helper methods like this to major/popular Solr classes, without first identifying that it's widely useful.

Reply:
Fair enough, this could be removed and put into the QueryComponent, since it's the only place that uses it.